### PR TITLE
[TASK] Support more file types in the `.editorconfig`

### PR DESCRIPTION
**Pull request by oliverklee**
Now we also have formatting options for YAML and JSON files.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -22,3 +22,13 @@ indent_size = 4
 [*.xml]
 indent_style = space
 indent_size = 4
+
+# YAML files
+[*.{yml,yaml}]
```
Comment (JakeQZ) [inline .editorconfig:27]: I can confirm this works with Notepad++ with EditorConfig plugin.

File: .editorconfig [hunk 2]
```diff
@@ -22,3 +22,13 @@ indent_size = 4
...
+indent_style = space
+indent_size = 2
+
+# JSON files
+[*.json]
+indent_style = space
+indent_size = 2
```
Comment (JakeQZ) [inline .editorconfig:34]: Our `composer.json` is indented with 4 spaces.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
Shouldn't it be 4 spaces indent for `.json`?

**Comment by JakeQZ** [comment]
Also noticed the 4 spaces indent already here for `.md`.  Aren't we indenting `.md` with 2 spaces (e.g. bulleted lists in README)?  Or have we not really decided?

File: .editorconfig (hunk 2 above)
Comment (oliverklee) [reply .editorconfig]: Changed an repushed.

**Comment by oliverklee** [comment]
I think we have not really decided. :-)

- https://www.markdownguide.org/basic-syntax/ does not say anything about indentation styles.
- https://markdown-guide.readthedocs.io/en/latest/basics.html says "4 spaces"
- https://google.github.io/styleguide/docguide/style.html also says 4 spaces

We might need to reformat our markdown files after this.

**Review by JakeQZ — APPROVED** [review]

**Pull request merged**